pkg/app/app.go

```diff
@@ -1182,14 +1182,24 @@ func (a *App) findDesiredStateFiles(specifiedPath string, opts LoadOpts) ([]stri
 		case defaultFile != "":
 			return []string{defaultFile}, nil
 		default:
-			return []string{}, fmt.Errorf("no state file found. It must be named %s/*.{yaml,yml} or %s, otherwise specified with the --file flag", DefaultHelmfileDirectory, DefaultHelmfile)
+			return []string{}, fmt.Errorf("no state file found. It must be named %s/*.{yaml,yml,yaml.gotmpl,yml.gotmpl} or %s, otherwise specified with the --file flag", DefaultHelmfileDirectory, DefaultHelmfile)
```

Comment:
This change looks reasonable. However, neither v0 or v1 seem to actually look for helmfile.yaml.gotmpl by default 🤔
Try running helmfile without any args- it won't find `helmfile.yaml.gotmpl`. SImilarly, try running helmfile with `-f somedir/` and put `somedir/helmfile.yaml.gotmpl`- it won't find the config.

Probably we'd need to fix those issues before or immediately after merging this one.